src/plugins/intel_cpu/src/nodes/subgraph.cpp

```diff
@@ -965,11 +992,34 @@ void Subgraph::SubgraphExecutor::parallel_forNd(const std::function<void(jit_sni
                 tmp /= dom[j];
             }
 
-            caller(call_args, indexes.data());
+            caller(call_args, indexes);
         }
     });
 }
 
+void Subgraph::SubgraphExecutor::execute(dnnl::stream strm, std::vector<MemoryPtr>& inMemPtrs, std::vector<MemoryPtr>& outMemPtrs) {
+    if (m_in_requested_descs.empty())
+        exec_impl(inMemPtrs, outMemPtrs);
+    else
+        reorder_execute(strm, inMemPtrs, outMemPtrs);
```

Comment:
What's about to split reordering and execution?
Smth like this:
```
if (need_prereordering()) {
    prereorder(strm, inMemPtrs);  // only reorder
}
exec_impl(inMemPtrs, outMemPtrs);

```

Reply:
The current approach allows to avoid `inMemPtrs` vector copy in case when prereordering is not needed (and this is the most popular case). However, I could overrate the impact of such unnecessary copy, so we can use the proposed solution

Reply:
Refactored, please take a look